internal/connector/internal/migrations/202302030000_add_processor_tables.go

```diff
@@ -0,0 +1,166 @@
+		ClusterID                string
+		NamespaceID              string
+		AllowUpgrade             bool
+		Status                   ProcessorDeploymentStatus `gorm:"foreignKey:ID;references:ID"`
+		Annotations              []ProcessorAnnotation     `gorm:"foreignKey:ProcessorID;references:ProcessorID"`
+	}
+
+	type LeaderLease struct {
+		db.Model
+		Leader    string
+		LeaseType string
+		Expires   *time.Time
+	}
+
+	return db.CreateMigrationFromActions(migrationId,
+		db.FuncAction(func(tx *gorm.DB) error {
+			// We don't want to delete the leader lease table on rollback because it's shared with the kas-fleet-manager
+			// so, we just create it here if it does not exist yet... but we don't drop it on rollback.
+			err := tx.Migrator().AutoMigrate(&LeaderLease{})
+			if err != nil {
+				return err
+			}
+			now := time.Now().Add(-time.Minute) //set to an expired time
+			return tx.Create(&api.LeaderLease{
+				Expires:   &now,
+				LeaseType: "processor",
+			}).Error
+		}, func(tx *gorm.DB) error {
+			// The leader lease table may have already been dropped, by the kafka migration rollback, ignore error
+			_ = tx.Where("lease_type = ?", "processor").Delete(&api.LeaderLease{})
+			return nil
+		}),
+		db.CreateTableAction(&ProcessorStatus{}),
+		db.CreateTableAction(&ProcessorAnnotation{}),
+		db.CreateTableAction(&Processor{}),
+		db.ExecAction(`                
+			CREATE OR REPLACE FUNCTION processors_version_trigger() RETURNS TRIGGER LANGUAGE plpgsql AS '
+			BEGIN
+			NEW.version := nextval(''processors_version_seq'');
+			RETURN NEW;
+			END;'
+		`, `
+			DROP FUNCTION processors_version_trigger
+		`),
+		db.ExecAction(`DROP TRIGGER IF EXISTS processors_version_trigger ON processors`, ``),
+		db.ExecAction(`
+			CREATE TRIGGER processors_version_trigger BEFORE INSERT OR UPDATE ON processors
+			FOR EACH ROW EXECUTE PROCEDURE processors_version_trigger();
+		`, `
+			DROP TRIGGER processors_version_trigger ON processors
+		`),
+		db.CreateTableAction(&ProcessorDeploymentStatus{}),
+		db.CreateTableAction(&ProcessorDeployment{}),
+		db.ExecAction(`
+			CREATE OR REPLACE FUNCTION processor_deployments_version_trigger() RETURNS TRIGGER LANGUAGE plpgsql AS '
+			BEGIN
+			NEW.version := nextval(''processor_deployments_version_seq'');
+			RETURN NEW;
+			END;'
+		`, `
+			DROP FUNCTION processor_deployments_version_trigger
+		`),
+		db.ExecAction(`DROP TRIGGER IF EXISTS processor_deployments_version_trigger ON processor_deployments`, ``),
+		db.ExecAction(`
+			CREATE TRIGGER processor_deployments_version_trigger BEFORE INSERT OR UPDATE ON processor_deployments
+			FOR EACH ROW EXECUTE PROCEDURE processor_deployments_version_trigger();
+		`, `
+			DROP TRIGGER processor_deployments_version_trigger ON processor_deployments
+		`),
+		db.CreateTableAction(&ProcessorShardMetadata{}),
+		db.FuncAction(func(tx *gorm.DB) error {
+			// Processors don't really support different metadata but include a default until we decide how we want to manage
```

Comment:
FYI. Support for _something_ for `processor`'s. Not sure what.. but it's supported :smile: